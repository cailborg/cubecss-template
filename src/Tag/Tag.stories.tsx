import { Story, Meta } from "@storybook/react/types-6-0";
import { Tag, TagProps } from "./Tag";

export default {
  title: "Components/Tag",
  component: Tag,
  argTypes: {
    label: { control: "text" },
  },
} as Meta;

export const All: Story<TagProps> = (args) => <Tag {...args}></Tag>;
All.args = { label: 'Button Text' };

