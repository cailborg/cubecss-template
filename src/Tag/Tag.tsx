import React from "react";
import styled from "@emotion/styled";

export interface TagProps {
    variant: "positive" | "neutral" | "warning" | "negative";
    size: "large" | "medium" | "small";
    label: string;
}

const getSize = (size: string) => {
    switch (size) {
        case "small":
            return `font-size: var(--lp-font-size-legal);
            font-weight: var(--lp-font-weight-legal-bold);
            line-height: var(--lp-line-height-legal);
            letter-spacing: var(--lp-letter-spacing-legal);
            padding: 2px 6px;`;
        case "medium":
            return `font-size: var(--lp-font-size-caption);
            font-weight: var(--lp-font-weight-caption-bold);
            line-height: var(--lp-line-height-caption);
            letter-spacing: var(--lp-letter-spacing-caption);
            padding: 4px 8px`;
        case "large":
            return `font-size: var(--lp-font-size-body-3);
            font-weight: var(--lp-font-weight-body-3-bold);
            line-height: var(--lp-line-height-body-3);
            letter-spacing: var(--lp-letter-spacing-body-3);
            padding: 4px 8px`;
        default:
            return `var(--lp-space-md)`;
    }
};

const StyledSpan = styled.span<{ size: string }>`
    display: inline-flex;
    border-radius: var(--lp-radii-lg);
    color: rgba(var(--lp-colors-neutral-800), 1);
    ${(props) => getSize(props.size)};
    &[data-variant="positive"] {
        background: rgba(var(--lp-colors-sky-500), 1);
    }
    &[data-variant="neutral"] {
        background: rgba(var(--lp-colors-neutral-200), 1);
    }
    &[data-variant="warning"] {
        background: rgba(var(--lp-colors-sunrise-500), 1);
    }
    &[data-variant="negative"] {
        background: rgba(var(--lp-colors-persimmon-500), 1);
    }
`;

export const Tag: React.FC<TagProps> = ({
    variant = "primary",
    size = "large",
    label,
}) => {
    return (
        <StyledSpan size={size} aria-live="polite" data-variant={variant}>
            {label}
        </StyledSpan>
    );
};
