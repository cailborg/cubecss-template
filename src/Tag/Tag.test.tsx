import React from "react";
import { Tag } from "../Tag/Tag";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Tag variant="neutral" size="large" label="hello"></Tag>).toJSON();
  expect(tree).toMatchSnapshot();
});
