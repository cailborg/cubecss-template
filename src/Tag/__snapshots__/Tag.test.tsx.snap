// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  aria-live="polite"
  className="css-ul6w35"
  data-variant="neutral"
  size="large"
>
  hello
</span>
`;
